python/test/unit/spacewalk/satellite_tools/test_yum_src.py

```diff
@@ -213,6 +214,85 @@ def test_mirror_list_arch(self):
                 "http://host3/base/arch1/os/",
             ])
 
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.initCFG", Mock())
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.os.unlink", Mock())
+    @patch("urlgrabber.grabber.PyCurlFileObject", Mock())
+    @patch("spacewalk.common.rhnLog", Mock())
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.fileutils.makedirs", Mock())
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.etree.parse", MagicMock(side_effect=Exception))
+    def test_proxy_usage_with_mirrorlist(self):
+        cs = self._make_dummy_cs()
+        proxy_url = "http://proxy.example.com:8080"
+        proxy_user = "user"
+        proxy_pass = "pass"
+        cs.proxy_hostname = proxy_url
+        cs.proxy_user = proxy_user
+        cs.proxy_pass = proxy_pass
+        expected_url_list = []
+        url_list = [
+            "http://example/base/arch1/os/",
+            "http://example/",
+            "http://example.com/",
+            "https://example.org/repo/path/?token",
+            ]
+        for url in url_list:
+            if "?" in url:
+                query = "&"
+            else:
+                query = "?"
+            expected_url_list.append("{}{}proxy={}&proxyuser={}&proxypass={}".format(url,
+                                                                                     query,
+                                                                                     quote(proxy_url),
+                                                                                     proxy_user,
+                                                                                     proxy_pass
+                                                                                     ))
+        grabber_mock = Mock()
+
+        with patch("spacewalk.satellite_tools.repo_plugins.yum_src.urlgrabber.urlgrab", grabber_mock):
+            with open(os.path.join(self.repo.root, "mirrorlist.txt"), "w") as fake_list:
```

Comment:
I would put the `mirrorlist.txt` into a [temporary directory ](https://docs.pytest.org/en/6.2.x/tmpdir.html#temporary-directories-and-files)that get's cleaned up by pytest afterwards

Reply:
I modeled this part after the existing test `test_mirror_list_arch`. So I'll change it in both tests to be consistent

Reply:
It seems like this doesn't work as easily in uyuni. Probably because we use an older version of pytest. At least my testing showed that all the fixtures return tuples instead of path or pathlike objects like the documentation says.